### PR TITLE
examples/cors: Fixed unable to open file '/var/log/access.log' in front_envoy

### LINKED ISSUE #11836: example/cors: Front envoy service unable to start in frontend and backend
Front envoy fails on both frontend and backend with the same log given below.

`front-envoy_1       | [2020-07-01 11:36:32.046][8][critical][main] [source/server/server.cc:101] error initializing configuration '/etc/front-en
voy.yaml': unable to open file '/var/log/access.log': Permission denied
front-envoy_1       | [2020-07-01 11:36:32.047][8][info][main] [source/server/server.cc:704] exiting
front-envoy_1       | unable to open file '/var/log/access.log': Permission denied`

### PR DESCRIPTION
**Pull request by mk46**
Signed-off-by: Manish Kumar <manish.kumar1@india.nec.com>
 
Commit Message: examples/cors: Fixed unable to open file '/var/log/access.log' in front_envoy
Additional Description: Running front_envoy container as root to fix this issue.  

Fixes #11836

### CODE REVIEW

**Comment by dio** [comment]
Hum. cc. @phlax 

Relevant: https://github.com/moby/moby/issues/31243, especially this comment: https://github.com/moby/moby/issues/31243#issuecomment-406879017.

**Comment by phlax** [comment]
@dio - why would it not work? 

it certainly seemed to work when i tested the image/example locally

i think this problem is because the user/container is trying to write to `/var/log/access.log` - not sure on the best fix here, other than chowning the `/var/log` dir

**Comment by mk46** [comment]
> @mk46 seems like you discovered that:
> 
> https://github.com/envoyproxy/envoy/blob/011945dcf92b8a461ab4ba309fa2bebeffc15895/ci/docker-entrypoint.sh#L24
> 
> doesn't work?
> Hum. cc. @phlax
> 
> Relevant: [moby/moby#31243](https://github.com/moby/moby/issues/31243), especially this comment: [moby/moby#31243 (comment)](https://github.com/moby/moby/issues/31243#issuecomment-406879017).

It's working for me. I have checked the log inside the container.

**Comment by dio** [comment]
@phlax sorry for looking at the wrong destination files. It is /var/log, seems like it doesn't make sense to write the log to /var/log/access.log inside a container. Probably change that to `/dev/stdout`?

**Comment by phlax** [comment]
i updated above comment - i can pr a doc fix shortly

File: examples/cors/backend/docker-compose.yaml [hunk 1]
```diff
@@ -7,6 +7,8 @@ services:
       dockerfile: Dockerfile-frontenvoy
     volumes:
       - ./front-envoy.yaml:/etc/front-envoy.yaml
+    environment:
+      ENVOY_UID: 0
```
Comment (dio) [inline examples/cors/backend/docker-compose.yaml:11]: Can we change the access log path to `/dev/stdout` instead?

```yaml
          access_log:
            - name: envoy.access_loggers.file
              typed_config:
                "@type": type.googleapis.com/envoy.config.accesslog.v2.FileAccessLog
                path: "/dev/stdout"
```

Comment (mk46) [reply examples/cors/backend/docker-compose.yaml:11]: I will do that but If the user will modify path other than "/dev/stdout" then the issue will be the same.

Comment (phlax) [reply examples/cors/backend/docker-compose.yaml:11]: yep, i think we should add some docs re log permissions for that reason

**Comment by dio** [comment]
@mk46 thanks, can you remove the following? 

```yaml
    environment:
      ENVOY_UID: 0  
```

**Comment by mk46** [comment]
> @mk46 thanks, can you remove the following?
> 
> ```yaml
>     environment:
>       ENVOY_UID: 0  
> ```

It fails with  below log 

`front-envoy_1      | [2020-07-01 16:03:36.955][7][critical][main] [source/server/server.cc:100] error initializing configuration '/etc/front-envoy.yaml': unable to open file '/dev/stdout': Permission denied
front-envoy_1      | [2020-07-01 16:03:36.955][7][info][main] [source/server/server.cc:692] exiting
front-envoy_1      | unable to open file '/dev/stdout': Permission denied`

**Comment by phlax** [comment]
@mk46 are you using the latest docker image ?

it may be that it hasnt been rebuilt/pushed yet

**Comment by mk46** [comment]
Yeah, I am using the latest image.

**Comment by phlax** [comment]
@dio @mk46 given that /dev/stdout is the default - perhaps the solution is to remove the log line altogether - not sure if its better to be explicit here

@mk46 you need to rebuild from master in that case - but to be sure try

```
docker pull envoyproxy/envoy-dev:latest
```
and rebuild. If that doesnt work then you will need the example recipe from master for `envoyproxy/envoy-dev`

**Comment by dio** [comment]
> given that /dev/stdout is the default - perhaps the solution is to remove the log line altogether - not sure if it's better to be explicit here

This is the access log path, not the "application" log https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/access_loggers/file/v3/file.proto#envoy-v3-api-field-extensions-access-loggers-file-v3-fileaccesslog-path

**Comment by mk46** [comment]
@dio @phlax Sorry, Something mesh with the dev-env. I tried on another machine, it's working without ENVOY_UID: 0. Thanks!

**Pull request merged**